policy/compressor_mock.go

```diff
@@ -0,0 +1,117 @@
+package policy
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/m3db/m3cluster/kv"
+	"github.com/m3db/m3metrics/generated/proto/schema"
+	"github.com/m3db/m3x/time"
+)
+
+// mockCompressionValue is a mock of a kv.Value that can be used for testing policy compression
+type mockCompressorValue struct {
+	policies compressorMap
+	version  int
+}
+
+func (m *mockCompressorValue) Unmarshal(v proto.Message) error {
+	activePolicies, ok := v.(*schema.ActivePolicies)
+	if !ok {
+		return fmt.Errorf("invalid proto.Message, expecting *schema.ActivePolicies, received %T", v)
+	}
+
+	for policy, id := range m.policies {
+		precision, err := xtime.DurationFromUnit(policy.resolution.Precision)
+		if err != nil {
+			return err
+		}
+		res := schema.Resolution{WindowSize: int64(policy.resolution.Window), Precision: int64(precision)}
+		ret := schema.Retention{Period: int64(policy.retention)}
+		p := schema.Policy{Resolution: &res, Retention: &ret}
+		e := schema.CompressedPolicy{Policy: &p, Id: id}
+
+		activePolicies.CompressedPolicies = append(activePolicies.CompressedPolicies, &e)
+	}
+
+	return nil
+}
+
+func (m *mockCompressorValue) Version() int {
+	return m.version
+}
+
+func (m *mockCompressorValue) IsNewer(other kv.Value) bool {
+	return m.version > other.Version()
+}
+
+// mockCompressionValueWatch is a mock of kv.ValueWatch that can be used for testing policy compression
+type mockCompressorValueWatch struct {
```

Comment:
Rather than making a MockValueWatch, could you try using the kv.ValueWatchable?

```
vw := kv.NewValueWatchable()
w := vw.Watch()
vw.Update(mockValue)
// you will be getting updates from w.C() and w.Get()
```